src/DsaTopDown.cc

```diff
@@ -99,9 +98,18 @@ bool TopDownAnalysis::runOnModule(Module &M, GraphMap &graphs) {
     postorder_scc.push_back(*it);
   }
 
+  const size_t totalFunctions = M.getFunctionList().size();
+  size_t functionsProcessed = 0;
+  BrunchTimer tdTimer("TD");
+
   for (auto it = postorder_scc.rbegin(), et = postorder_scc.rend(); it != et;
        ++it) {
     auto &scc = *it;
+
+    SEA_DSA_BRUNCH_PROGRESS("TD_FUNCTIONS_PROCESSED", functionsProcessed,
+                            totalFunctions);
+    functionsProcessed += scc.size();
```

Comment:
Ouch. the stats all over the place is very annoying :(

Reply:
it's enabled by a separate flag, `--log=brunch-progress`, so shouldn't bother anyone.
we can also print it more rarely, say every 1% of progress or something

Reply:
I mean code for collecting and managing stats is all over the place in the code. Of course I see how this is controlled at runtime. I'm annoyed by code bloat not performance.

Reply:
Also in the current code, `BRUNCH_STAT` is restricted to `Stats` only and the magic string is not spread out throughout the code (I hope ;) )